### PR TITLE
feat: Disable notif on login and onboarding

### PR DESCRIPTION
**Pull request by CrownKira**
## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

File: frontend/src/components/guards/CommonGuard.tsx [hunk 1]
```diff
@@ -16,23 +18,27 @@ interface Props {
...
+  const checkNotAuth = useCheckNotAuth();
   const checkOnboarded = useCheckOnboarded();
   const checkNotOnboarded = useCheckNotOnboarded();
 
   const guards = [];
   authenticated && guards.push(checkAuth);
+  notAuthenticated && guards.push(checkNotAuth);
   onboarded && guards.push(checkOnboarded);
-  notOnboarded && guards.push(checkNotOnboarded);
+  notOnboarded && guards.push(() => checkNotOnboarded(true, true));
```
Comment (therizhao) [inline frontend/src/components/guards/CommonGuard.tsx:41]: Nit: Prefer named args for check not onboarded

File: frontend/src/hooks/guards/useCheckNotAuth.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import useAuthProvider from '../auth/useAuthProvider';
+
+type CheckNotAuth = (redirectTo?: string) => Promise<any>;
+
+const useCheckNotAuth = (): CheckNotAuth => {
+  const authProvider = useAuthProvider();
+  const navigate = useNavigate();
+
+  const checkNotAuth = useCallback(
+    async (redirectTo = PATH_DASHBOARD.root) => {
```
Comment (therizhao) [inline frontend/src/hooks/guards/useCheckNotAuth.ts:13]: ^ Same as above Prefer named args

File: frontend/src/routes/index.tsx [hunk 3]
```diff
@@ -56,7 +60,7 @@ export default function Router() {
       element: <Page404 />,
     },
     { path: '/', element: <Navigate to="/dashboard?disableNotification=true" replace /> },
```
Comment (therizhao) [inline frontend/src/routes/index.tsx:62]: Oops I accidentally added this in my commit. Can help to remove ty.

**Review by therizhao — APPROVED** [review]
One Minor fix required. I just approve first.

Can merge when done.

File: frontend/src/components/guards/CommonGuard.tsx (hunk 1 above)
Comment (CrownKira) [reply frontend/src/components/guards/CommonGuard.tsx]: Done!

File: frontend/src/routes/index.tsx (hunk 3 above)
Comment (CrownKira) [reply frontend/src/routes/index.tsx]: Sure!

**Pull request merged**